cmd/sonobuoy/app/operations/gen_plugin.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operations
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/loader"
+)
+
+const (
+	placeholderHostname  = "<hostname>"
+	placeholderNamespace = "sonobuoy"
+)
+
+// GenPluginConfig are the input options for running
+type GenPluginConfig struct {
+	Paths      []string
+	PluginName string
+	// for testing
+	outfile io.Writer
```

Comment:
This is a bit of a smell to me -- I think GeneratePluginManifest shouldn't worry about where it is writing bytes to. If the return type were ([]byte, error) then the writing logic could move up into the command code. 

The test for the command code could become an Example where we could capture/compare Stdout and tests for GeneratePluginManifest wouldn't have to worry about providing a Writer to the config.

Reply:
I agree we should serialize to an `byte[]` which then can be processed by anything.